tests/console/zfs.pm

```diff
@@ -0,0 +1,190 @@
+    select_console 'root-console';
+}
+
+sub run {
+    # Preparation
+    my $self = shift;
+    # TODO: Switch to serial_terminal for BLAZING FAST EXECUTING!!!
+    #$self->select_serial_terminal;
+    select_console 'root-console';
+    install_zfs();
+    assert_script_run('modprobe zfs');
+    prepare_disks();
+    # Create zfs test pools
+    assert_script_run('zpool create tank raidz /var/tmp/tank_{a,b,c}.img -o ashift=13');
+    assert_script_run('zpool list | grep tank');
+    assert_script_run('zpool create dozer /var/tmp/dozer.img -o ashift=12');
+    assert_script_run('zpool list | grep dozer');
+    assert_script_run('cd /tank/; ls');
+    # Put some data on tank
+    assert_script_run('curl -v -o /tank/Big_Buck_Bunny_8_seconds_bird_clip.ogv ' . data_url('Big_Buck_Bunny_8_seconds_bird_clip.ogv'));
+    assert_script_run('md5sum Big_Buck_Bunny_8_seconds_bird_clip.ogv > /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    scrub('tank');
+    assert_script_run('zpool status tank | grep state | grep ONLINE');
+    script_run('zpool status tank');
+    save_screenshot();
+    ## Test raidz capability
+    corrupt_disk('/var/tmp/tank_a.img');
+    # Wait until degraded
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    scrub('tank');
+    assert_script_run("zpool status tank | grep scan | grep 'scrub repaired'");
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    script_run('zpool status tank');
+    assert_script_run('zpool status tank | grep "corrupted data"');
+    # Replace disk
+    assert_script_run('zpool offline tank /var/tmp/tank_a.img');
+    assert_script_run('zpool replace tank /var/tmp/tank_a.img');
+    scrub('tank');
+    assert_script_run('zpool status tank | grep state | grep ONLINE');
+    # Display status for debugging purposes
+    script_run('zpool status tank');
+    save_screenshot();
+    ## Test Snapshots
+    assert_script_run('zfs snapshot tank@initial');
+    assert_script_run('zfs list -t snapshot | grep "tank@initial"');
+    assert_script_run('curl -v -o /tank/test_unzip.zip ' . data_url('console/test_unzip.zip'));
+    assert_script_run('rm /tank/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    # Create second snapshot with other data
+    assert_script_run('zfs snapshot tank@second');
+    assert_script_run('zfs snapshot tank@third');
+    assert_script_run('zfs list -t snapshot | grep "tank@second"');
+    assert_script_run('zfs list -t snapshot | grep "tank@third"');
+    # Mount snapshots
+    assert_script_run('mkdir -p /mnt/tank/{initial,second}');
+    assert_script_run('mount -t zfs tank@initial /mnt/tank/initial');
+    assert_script_run('mount -t zfs tank@second /mnt/tank/second');
+    assert_script_run('stat /mnt/tank/initial/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
```

Comment:
I'd try to do also `md5sum` to verify the file is still the same.

Reply:
Good idea! I've updated that on all checks.